metrics/events.go

```diff
@@ -110,11 +111,11 @@ func (m *Metrics) HandleCandidatesFilteredEvent(ctx context.Context, id types.Re
 	}
 }
 
-func (m *Metrics) HandleTimeToFirstByteEvent(ctx context.Context, id types.RetrievalID, eventTime time.Time) {
+func (m *Metrics) HandleTimeToFirstByteEvent(ctx context.Context, id types.RetrievalID, storageProviderId string, eventTime time.Time) {
 	tempData := m.tempDataMap.GetOrCreate(id)
 	if tempData.RecordTimeToFirstByte(eventTime) {
-		m.requestWithFirstByteReceivedCount.Add(ctx, 1)
-		m.timeToFirstByte.Record(ctx, eventTime.Sub(tempData.StartTime()).Seconds())
+		m.requestWithFirstByteReceivedCount.Add(ctx, 1, attribute.String("sp_id", storageProviderId), attribute.String("protocol", protocol(storageProviderId)))
+		m.timeToFirstByte.Record(ctx, eventTime.Sub(tempData.StartTime()).Seconds(), attribute.String("sp_id", storageProviderId), attribute.String("protocol", protocol(storageProviderId)))
```

Comment:
Not sure we want to track sp_ids for TTFB for TTLB, unless it's only for graphsync. My assumption is that the number of SPs << then number of IPFS providers. 

I'd mark this as a TODO, if we do need these stats later. Or, only add it if protocol = graphsync.